review

msharafjr opened the pull request.

msharafjr committed:
add index.js empty main file

msharafjr committed:
init package.json and gitignore

msharafjr committed:
add webpack configuration file

msharafjr committed:
setup server main file and routes file

msharafjr committed:
add index and signup components

msharafjr committed:
add users profile pics

msharafjr committed:
add js, css, fonts, stylesheets and views of public directory

msharafjr committed:
add login signup forms components

msharafjr committed:
add user registeration form component

msharafjr committed:
add user login form component

msharafjr committed:
add login and signup routes

msharafjr committed:
add home and signup css

msharafjr committed:
add navbar component

msharafjr committed:
add routes

msharafjr committed:
create some html files

msharafjr committed:
add new css files, rename forms to navbar and modify others

msharafjr committed:
add search jsx file

msharafjr committed:
add create-team jsx file

msharafjr committed:
modify signup jsx and separate signup-form file to 2 components for "user" and "admin"

msharafjr committed:
add post, put and delete requests

msharafjr committed:
add regex helper file

msharafjr committed:
delete all images and make one image for all users. pretty cool ha!

msharafjr committed:
add users.jsx file

msharafjr committed:
add fetch users

msharafjr committed:
remove alerts from addPlayer and use errorHandler function instead

msharafjr committed:
update signup form

msharafjr committed:
add team.jsx file

msharafjr committed:
modify creat-team

msharafjr committed:
add fetch for user's team

msharafjr committed:
add get-user, get-team and add-player routes

msharafjr committed:
add error handler middleware

msharafjr committed:
add minified js files, css and html

msharafjr committed:
modify index.jsx and add one regex helper

msharafjr committed:
add remove player functionality

msharafjr committed:
get players of the team

msharafjr committed:
modify team.jsx file

add admin page and add removePlayer function

msharafjr committed:
add search-team-box.jsx file

msharafjr committed:
add some components to webpack.config and modify package.json

msharafjr committed:
modify all team files and add team.css

msharafjr committed:
add room routes

msharafjr committed:
transitioning to single page application (SPA)

msharafjr committed:
add all components of team directory

msharafjr committed:
add loading component & css

shadyvb commented on controllers/components/index/login-form.jsx:
```diff
@@ -0,0 +1,184 @@
+            displayError: ''
+        };
+
+        this._checkEmptyFields = this._checkEmptyFields.bind( this );
```
This doesn't look right, what are you trying to fix ?

msharafjr replied:
Nothing, forgot to delete this, my bad.

shadyvb commented on controllers/components/index/login-form.jsx:
```diff
@@ -0,0 +1,184 @@
+        const name = target.name;
+
+        this.setState({
+            [ name ]: value
```
How can an array key be an .. array ?!

msharafjr replied:
[ES6 Computed property names syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Computed_property_names)
also in [React docs](https://facebook.github.io/react/docs/forms.html#handling-multiple-inputs)

shadyvb replied:
Interesting.. okay

shadyvb commented on controllers/components/index/login-form.jsx:
```diff
@@ -0,0 +1,184 @@
+        }
+
+        if (  emptyFeildsNames.length ) {
+            if ( emptyFeildsNames.length === 1 && emptyFeildsNames[ 0 ] == 'Email' ) {
```
DRY this code please.

msharafjr replied:
hahaha, gat mnen feilds de. mesh ana.

shadyvb replied:
DRY as in `Don't repeat yourself`

shadyvb commented on controllers/components/navbar.jsx:
```diff
@@ -0,0 +1,114 @@
+
+    _handleSearch( e ) {
+        const form = document.forms[ 's' ];
+        const search_for = prompt( 'Search for?\ntype: \'user\' or \'team\'' );
```
Use double quotes here so you can use single quotes without escaping.

msharafjr replied:
Yeh, good call.

shadyvb commented on controllers/components/navbar.jsx:
```diff
@@ -0,0 +1,114 @@
+                e.preventDefault();
+            }
+        } else {
+            e.preventDefault();
```
Why don't you do this outside the condition if you're doing it everywhere already ?

shadyvb commented on controllers/components/search/search.jsx:
```diff
@@ -0,0 +1,84 @@
+    }
+    componentWillMount() {
+        const that = this;
+        function getQueriesFromURL( query, url ) {
```
Do not ever, never, define functions within react lifecycle event handlers, it'll be defined each time the function is run.

msharafjr replied:
When a component renders, Doesn't `componentWillMount()` runs only once, before `render` gets called?
The point is it doesn't matter whether defining functions within it or not, it will run just once.
am right, or by convention it shouldn't?

shadyvb replied:
Be convention you shouldn't, yes. And the component might render multiple times, so that's why. I'm not entirely sure if the mounting happens on each render though. Still, better to define it as a separate function.

shadyvb commented on controllers/components/teams/team.jsx:
```diff
@@ -0,0 +1,218 @@
+
+    _identifyUser() {
+        if ( localStorage.token ) {
+            const user = JSON.parse( atob( localStorage.token.split( '.' )[ 1 ] ) );
```
Stop doing that all over, do it via a helper function somewhere.

msharafjr replied:
I put it as a helper function beside regex in `helpers.js`, will use it while changing most of components because i changed the payload in token.

shadyvb replied:
Thank you

shadyvb commented on controllers/components/users/users.jsx:
```diff
@@ -0,0 +1,217 @@
+                return '';
+            }
+        } else {
+            return '';
```
Really?! This looks ugly, fix it. ( identical code )

msharafjr replied:
Pyramid of doom 😄 
i will build the whole component from scratch.

shadyvb reviewed (commented):
Great work! Glad to see you took hold of React this good.

Got some feedback, ping me when you're done.

Also don't commit minified files, build process should do that for you. Just add docs on how to build and how to run the whole app.

shadyvb commented:
Do not add commits to an open PR :@

msharaf10 commented:
@shadyvb am done. Commit changes?

shadyvb commented:
@msharaf10 yes please, when you're done with all changes, ping me for another review.

msharafjr committed:
commit changes from PR#2

remove all token check statements, clean login & signup forms and remove declared functions from react lifecycle methods

msharaf10 commented:
@shadyvb changes committed.
